cmd/goQuery/cmd/root.go

```diff
@@ -99,6 +99,7 @@ func init() {
 	flags.BoolVarP(&cmdLineParams.Version, "version", "v", false, "Print version information and exit\n")
 
 	flags.StringVarP(&cmdLineParams.Ifaces, "ifaces", "i", "", helpMap["Ifaces"])
+	flags.StringVarP(&cmdLineParams.IfaceRegExp, "ifaces-re", "", "", helpMap["Ifaces-re"])
```

Comment:
We were talking about this for a while. It seemed like the easiest solution to stay backwards compatible, yet provide additional functionality.

The main problem being that the delimiter `,` used in the `--ifaces` parameter can be part of a regex, e.g. `eth{0,2,4}`

Reply:
Why not just follow the relatively common convention of enclosing any regex via `/.../` to indicate it is a regex? This can't conflict with an interface list (or any interface name for that matter) while still preventing the additional complexity.

Reply:
Well, can't say I have any strong feelings about this, but I think what ever is chosen it needs to be explicit. I'm not familiar with that convention, but then again go programmers seem to have a lot of conventions I would consider weird ;-)

I 
I don't really see it as avoiding complexity, either way the user would need to know exactly what he is doing by using different argument or wrapping the regex to backslashes. Implementation -wise also nota big difference, either we handle a separate argument or we need to check for backslashes and extract regex from the there. But as I have never really used the tool, I'm probably not the best at guessing what makes sense for the users.

Reply:
I don't think that convention is Go-specific, I've seen several tools / UIs that use the enclosing slashes as indicator that a regex is intended. That being said, I kinda have a strong opinion on this one (because IMHO the CLI is already not the most user-friendly in the world and quite cluttered). I agree that the user would need to know that there's a specific notation, but at least from my point of view it's more intuitive to be consistent in the `-i` parameter. Adding two `//` around the query also involves less typing.

If there are no strong objections from other sides I'd vote for this approach. @els0r ?

Reply:
He does not seem to have too strong feelings about this so I'll implement the wrapping the regexp to forward slashes. It will be fine.